ciao-controller/compute.go

```diff
@@ -461,24 +461,24 @@ func showServerDetails(w http.ResponseWriter, r *http.Request, context *controll
 	dumpRequest(r)
 
 	if validateToken(context, r) == false {
-		http.Error(w, "Invalid token", http.StatusInternalServerError)
+		http.Error(w, "Invalid token", http.StatusUnauthorized)
 		return
 	}
 
 	instance, err := context.ds.GetInstance(instanceID)
 	if err != nil {
-		http.Error(w, "Instance not available", http.StatusInternalServerError)
+		http.Error(w, "{\"itemNotFound\": {\"message\": \"Instance could not be found.\", \"code\": 404}", http.StatusNotFound)
 		return
 	}
 
 	if instance.TenantID != tenant {
-		http.Error(w, "Instance not available", http.StatusInternalServerError)
+		http.Error(w, "{\"itemNotFound\": {\"message\": \"Instance not in tenant.\", \"code\": 404}", http.StatusNotFound)
 		return
 	}
 
 	server.Server, err = instanceToServer(context, instance)
 	if err != nil {
-		http.Error(w, "Instance not available", http.StatusInternalServerError)
+		http.Error(w, "{\"itemNotFound\": {\"message\": \"Instance could not be found.\", \"code\": 404}", http.StatusNotFound)
```

Comment:
@leoswaldo Please add compute API error payloads to payloads/compute.go and use them to generate JSO payload through marshaling.